### PR TITLE
change vmap tests to dtypes=OpDTypes.any_one

### LINKED ISSUE #1092: Make vmap tests use dtype `any_one`
In #1069, @kshitij12345 smartly pointed out that it's disturbing that these batch rules aren't caught by `test_op_has_batch_rule`. From looking at it, the bitwise ops in particular aren't being tested because the only allowed_dtype is `torch.float`

### Steps
1. First, please update both [`test_vmap`](https://github.com/pytorch/pytorch/blob/master/test/functorch/test_vmap.py#L3501) and [`test_op_has_batch_rule`](https://github.com/pytorch/pytorch/blob/master/test/functorch/test_vmap.py#L3668) to have their `allowed_dtypes` (in the `@ops` decorator) be [`OpDTypes.any_one`](https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_device_type.py#L747-L748) instead of `torch.float32`
2. We expect this to lead to new failures. Please update the corresponding xfail list for the test.
    i. In the case of `test_op_has_batch_rule`, if the failure looks to occur on an in-place function, please try first to **only** add it the [`inplace_failures`](https://github.com/pytorch/pytorch/blob/master/test/functorch/test_vmap.py#L3670) list. If this does not work, you can xfail it

### PR DESCRIPTION
**Pull request by srossross**
Fixes https://github.com/pytorch/functorch/issues/1092

### CODE REVIEW

File: test/functorch/test_vmap.py [hunk 1]
```diff
@@ -3497,6 +3500,9 @@ def test():
         # The error inputs are vectors, that pass when batched as they are treated as a matrix
         xfail('trace'),
         xfail('as_strided', 'partial_views'),
+        xfail('nn.functional.one_hot'),  # please provide an explicit positive num_classes argument.
+        xfail('tril_indices'),  # Expected at least one Tensor to vmap over
+        xfail('triu_indices'),  # Expected at least one Tensor to vmap over
```
Comment (srossross) [inline test/functorch/test_vmap.py:3505]: This appears to be caused by the `sample_inputs_*` arguments.

```
In [31]: vmap(torch.nn.functional.one_hot)(x, num_classes=10)
Out[31]: 
tensor([[[1, 0, 0, 0, 0, 0, 0, 0, 0, 0], ...

```

File: test/functorch/test_vmap.py [hunk 2]
```diff
@@ -3680,6 +3691,14 @@ def test_op_has_batch_rule(self, device, dtype, op):
             'atan',
             'atanh',
             'baddbmm',
+            'bitwise_and',
+            'bitwise_or',
+            'bitwise_xor',
+            'bitwise_left_shift',
+            'bitwise_right_shift',
+            'gcd',
+            'lcm',
```
Comment (srossross) [inline test/functorch/test_vmap.py:3700]: added to `inplace_failures`

**Comment by srossross** [comment]
already fixed in https://github.com/pytorch/pytorch/pull/91355

**Comment by kshitij12345** [comment]
Sorry I forgot to reference that issue in #91355

**Pull request closed without merging**